### PR TITLE
frontend: status use get and set config internally

### PR DESCRIPTION
**Pull request by thisconnect**
Refactored status component so it uses setConfig and getConfig from utils, as setConfig keeps a temporary config so that successive calls don't loose previous changes.

Also changed to pass getConfig from utils/config for convenience.

### CODE REVIEW

**Comment by thisconnect** [comment]
first commit has a aopp.ts:63 Uncaught ReferenceError: Cannot access 'getConfig' before initialization that I think I could point down to https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/i18n/i18n.ts#L109

File: frontends/web/src/utils/config.ts [hunk 1]
```diff
@@ -25,12 +25,21 @@ type TConfig = {
...
 /**
- * expects an object with a backend or frontend key
- * i.e. { frontend: { language }}
+ * get current configs
+ * i.e. await getConfig()
+ * returns a promise with backend and frontend configs
+ */
+export const getConfig = (): Promise<any> => {
+  return apiGet('config');
+};
```
Comment (thisconnect) [inline frontends/web/src/utils/config.ts:34]: Ideally we wouldn't use any for config, I tired with unknown.. something like following, but ran into too many issues. 

```diff
diff --git a/frontends/web/src/utils/config.ts b/frontends/web/src/utils/config.ts
index 581325412..f4cbae175 100644
--- a/frontends/web/src/utils/config.ts
+++ b/frontends/web/src/utils/config.ts
@@ -18,18 +18,22 @@
 import { apiGet, apiPost } from '../utils/request';
 
 type TConfig = {
-  backend?: unknown;
-  frontend?: unknown;
+  backend: {
+    [key: string]: unknown
+  };
+  frontend: {
+    [key: string]: unknown
+  };
 };
 
-let pendingConfig: TConfig = {};
+let pendingConfig: Partial<TConfig> = {};
 
 /**
  * get current configs
  * i.e. await getConfig()
  * returns a promise with backend and frontend configs
  */
-export const getConfig = (): Promise<any> => {
+export const getConfig = (): Promise<TConfig> => {
   return apiGet('config');
 };
 
@@ -38,9 +42,9 @@ export const getConfig = (): Promise<any> => {
  * i.e. await setConfig({ frontend: { language }})
  * returns a promise and passes the new config
  */
-export const setConfig = (object: TConfig) => {
+export const setConfig = (object: Partial<TConfig>) => {
   return getConfig()
-    .then((currentConfig = {}) => {
+    .then((currentConfig) => {
       const nextConfig = Object.assign(currentConfig, {
         backend: Object.assign({}, currentConfig.backend, pendingConfig.backend, object.backend),
         frontend: Object.assign({}, currentConfig.frontend, pendingConfig.frontend, object.frontend)
```

**Comment by thisconnect** [comment]
rebased

**Review by shonsirsha — APPROVED** [review]
lgtm 👍

**Pull request merged**